### ISSUE TITLE
Pylint Unable to create directory /github/home/.cache/pylint

### ISSUE THREAD

**Issue by Kurt-von-Laven** [github/super-linter#1919]
## Describe the bug

Pylint itself logs what appears to be a permissions error when it detects a lint error. 

## Expected behavior

Pylint successfully caches stats files rather than failing to write to the file system.

## Steps to Reproduce

1. Create a Python file named foo.py containing a Pylint violation, such as a long line.
2. Create an empty pyproject.toml.
3. Run Super-Linter slim-v.4.7.1 on the file with the following options:

```yaml
          LINTER_RULES_PATH: /
          PYTHON_PYLINT_CONFIG_FILE: pyproject.toml
```

## Logs

```
Unable to create directory /github/home/.cache/pylint
Unable to create file /github/home/.cache/pylint/foo1.stats: [Errno 2] No such file or directory: '/github/home/.cache/pylint/foo1.stats'
```

**Comment by admiralAwkbar** [comment]
@Kurt-von-Laven Would it make sense to when we build the image we create the folder and have it open?

**Comment by Kurt-von-Laven** [comment]
That might fix the issue, but I don't recommend that approach off the bat. This strikes me as a permissions issue, so these are the sorts of things I would look into.

- What is the effective user of the Pylint process (`top | grep pylint`)?
- What group is that user in (`groups <username>`)?
- What permissions/ownership settings are set on the `/github/home/` directory (`ls -l`)?
- Does a `/github/home/.cache/` directory exist?

If all of the permissions look right, but there is no `~/.cache`, maybe this is instead a Pylint bug, and they need to be using `mkdir -p ~/.cache/pylint` rather than `mkdir ~/.cache/pylint`.

**Comment by admiralAwkbar** [comment]
@Kurt-von-Laven Thanks for the input. I need to get some time to test this.
Its been crazy week out here and any PRs are very welcome :)
As always, thanks for your insight and great issues

**Comment by Kurt-von-Laven** [comment]
@admiralAwkbar, this one has had me stumped for a while, but I recently hit [a permission error in a different context that was related to Docker running as root](https://github.com/actions/runner/issues/434). I wonder if that is also going on here and could be addressed by simply [running Docker in rootless mode](https://github.com/ScribeMD/rootless-docker), which also has security benefits. That repo's README also explains the other error.

**Comment by Kurt-von-Laven** [comment]
FYI, the job to remove the stale label failed.

**Comment by tunetheweb** [comment]
> FYI, the job to remove the stale label failed.

Fix for that proposed in #2032

**Comment by Kurt-von-Laven** [comment]
Thank you, @tunetheweb (and @admiralAwkbar for the marknotstale label).
